src/rust/catcollar/runtime/memory.rs

```diff
@@ -21,53 +20,60 @@ use crate::runtime::{
 use ::libc::c_void;
 use ::std::{
     mem,
-    slice,
+    ptr::{
+        self,
+        NonNull,
+    },
 };
 
 //==============================================================================
 // Trait Implementations
 //==============================================================================
 
-/// Memory Runtime Trait Implementation for I/O User Ring Runtime
+/// Memory Runtime Trait Implementation for IoUring Runtime
 impl MemoryRuntime for IoUringRuntime {
-    /// Creates a scatter-gather array from a memory buffer.
-    fn into_sgarray(&self, buf: Buffer) -> Result<demi_sgarray_t, Fail> {
-        let len: usize = buf.len();
-        #[allow(unreachable_patterns)]
-        let (dbuf_ptr, sgaseg): (*const u8, demi_sgaseg_t) = match buf {
-            Buffer::Heap(dbuf) => {
-                let (dbuf_ptr, data_ptr): (*const u8, *const u8) = DataBuffer::into_raw_parts(Clone::clone(&dbuf))?;
-                (
-                    dbuf_ptr,
-                    demi_sgaseg_t {
-                        sgaseg_buf: data_ptr as *mut c_void,
-                        sgaseg_len: len as u32,
-                    },
-                )
-            },
-            _ => return Err(Fail::new(libc::EINVAL, "invalid buffer type")),
+    /// Converts a buffer into a scatter-gather array.
+    fn into_sgarray(&self, buf: DemiBuffer) -> Result<demi_sgarray_t, Fail> {
+        // Create a scatter-gather segment to expose the DemiBuffer to the user.
+        let data: *const u8 = buf.as_ptr();
+        let sga_seg: demi_sgaseg_t = demi_sgaseg_t {
+            sgaseg_buf: data as *mut c_void,
+            sgaseg_len: buf.len() as u32,
         };
+
+        // Create and return a new scatter-gather array (which inherits the DemiBuffer's reference).
         Ok(demi_sgarray_t {
-            sga_buf: dbuf_ptr as *mut c_void,
+            sga_buf: buf.into_raw().as_ptr() as *mut c_void,
             sga_numsegs: 1,
-            sga_segs: [sgaseg],
+            sga_segs: [sga_seg],
             sga_addr: unsafe { mem::zeroed() },
         })
     }
 
     /// Allocates a scatter-gather array.
     fn alloc_sgarray(&self, size: usize) -> Result<demi_sgarray_t, Fail> {
-        // Allocate a heap-managed buffer.
-        let dbuf: DataBuffer = DataBuffer::new(size)?;
-        let (dbuf_ptr, data_ptr): (*const u8, *const u8) = DataBuffer::into_raw_parts(dbuf)?;
-        let sgaseg: demi_sgaseg_t = demi_sgaseg_t {
-            sgaseg_buf: data_ptr as *mut c_void,
+        // ToDo: Allocate an array of buffers if requested size is too large for a single buffer.
```

Comment:
I know that we have not agreed on a standard on how to name special comments such as TODOs, FIXMEs and so on. For instance, I've personally been using this latter syntax.

Let us discuss on this with the team, so that we be consistent.

Note that I don't take this this a non-blocker for the PR.

Reply:
Ah yes, this should probably be part of the coding standards document.

Reply:
@anandbonde could you note this down for out next discussion?

Reply:
I added a description of this to the "Coding Standards" Channel in Teams and asked Anand to add it to the agenda for the next meeting.